src/mscorlib/shared/System/IO/UnmanagedMemoryStream.cs

```diff
@@ -764,7 +779,16 @@ public override Task WriteAsync(ReadOnlyMemory<byte> source, CancellationToken c
 
             try
             {
-                Write(source.Span);
+                // See corresponding comment in ReadAsync for why we don't just always use Write(ReadOnlySpan<byte>).
+                // Unlike ReadAsync, we could delegate to WriteAsync(byte[], ...) here, but we don't for consistency.
+                if (source.DangerousTryGetArray(out ArraySegment<byte> sourceArray))
+                {
+                    Write(sourceArray.Array, sourceArray.Offset, sourceArray.Count);
+                }
+                else
+                {
+                    Write(source.Span);
+                }
```

Comment:
Why do you use the ? : syntax above but if-else here?

Reply:
I can't use it here.  Write is void-returning.

Reply:
ah, missed that.